### PR TITLE
doc: update README for C samples, add comments

### PR DESCRIPTION
**Pull request by generalova-kate**
### Details:
 - Update README with a new format
 - Add doc strings for functions and code blocks

### CODE REVIEW

File: inference-engine/ie_bridges/c/samples/hello_classification/main.c [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
 
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
-#include <opencv_c_wraper.h>
+
 #include <c_api/ie_c_api.h>
+#include <opencv_c_wraper.h>
 
+/**
```
Comment (mryzhov) [inline inference-engine/ie_bridges/c/samples/hello_classification/main.c:12]: Are we decided to use Javadoc style in addition to Google Google C++ Style?

Comment (generalova-kate) [reply inference-engine/ie_bridges/c/samples/hello_classification/main.c:12]: The style will be chosen and checked later with automatic pipeline.

File: inference-engine/ie_bridges/c/samples/hello_classification/README.md [hunk 2]
```diff
@@ -1,31 +1,102 @@
...
+> * Before running the sample with a trained model, make sure the model is converted to the Inference Engine format (\*.xml + \*.bin) using the [Model Optimizer tool](../../../../../docs/MO_DG/Deep_Learning_Model_Optimizer_DevGuide.md).
 >
-> The sample accepts models in ONNX format (.onnx) that do not require preprocessing.
+> * The sample accepts models in ONNX format (.onnx) that do not require preprocessing.
 
 You can do inference of an image using a trained AlexNet network on a GPU using the following command:
 
 ```sh
-./hello_classification_c <path_to_model>/alexnet_fp32.xml <path_to_image>/cat.bmp GPU
+./hello_classification_c <path_to_model>/alexnet_fp32.xml <path_to_image>/cat.png GPU
```
Comment (mryzhov) [inline inference-engine/ie_bridges/c/samples/hello_classification/README.md:53]: bmp is the universal format for all our samples, png cannot be open without OpenCV. But this one supports it.

File: inference-engine/ie_bridges/c/samples/object_detection_sample_ssd/c_w_dirent.h [hunk 3]
```diff
@@ -96,13 +128,18 @@ static DIR *opendir(const char *dirPath) {
     dp->hFind = FindFirstFileA(ws, &dp->FindFileData);
     dp->FindFileData.dwReserved0 = dp->hFind != INVALID_HANDLE_VALUE;
     free(ws);
-    if (isValid(dp)) {
+    if (!isValid(dp)) {
```
Comment (mryzhov) [inline inference-engine/ie_bridges/c/samples/object_detection_sample_ssd/c_w_dirent.h:131]: Could you please fix the functional issues in separate PR? looks like we have no tests for this case.

File: inference-engine/ie_bridges/c/samples/object_detection_sample_ssd/main.c [hunk 4]
```diff
@@ -69,11 +76,15 @@ int ParseAndCheckCommandLine(int argc, char *argv[]) {
...
-        printf("Model is required but not set. Please set -m option. \n");
+        printf("Model is required but not set. Please set -m option.\n");
         return -1;
     }
     if (img_msg == NULL) {
-        printf("Input is required but not set.Please set - i option.\n");
+        printf("Input is required but not set.Please set -i option.\n");
+        return -1;
+    }
+    if (custom_plugin_cfg_msg == NULL && device_name != "CPU") {
```
Comment (mryzhov) [inline inference-engine/ie_bridges/c/samples/object_detection_sample_ssd/main.c:86]: Looks like the sample cannot be used for GPU without custom kernels anymore

File: inference-engine/ie_bridges/c/samples/hello_nv12_input_classification/main.c [hunk 5]
```diff
@@ -81,7 +111,14 @@ size_t read_image_from_file(const char *img_path, unsigned char *img_data, size_
...
 
-size_t parse_image_size(const char *size_str, size_t *width, size_t *height) {
+/**
+* @brief Get image width and height
+* @param string image size in WIDTHxHEIGHT format
+* @param pointer to image width
+* @param pointer to image height
+* @return int status 0(success) or -1(fail)
+*/
+int parse_image_size(const char *size_str, size_t *width, size_t *height) {
```
Comment (mryzhov) [inline inference-engine/ie_bridges/c/samples/hello_nv12_input_classification/main.c:121]: short int? But I think bool is more suitable here

File: inference-engine/ie_bridges/c/samples/object_detection_sample_ssd/c_w_dirent.h (hunk 3 above)
Comment (generalova-kate) [reply inference-engine/ie_bridges/c/samples/object_detection_sample_ssd/c_w_dirent.h]: move it

File: inference-engine/ie_bridges/c/samples/object_detection_sample_ssd/main.c (hunk 4 above)
Comment (generalova-kate) [reply inference-engine/ie_bridges/c/samples/object_detection_sample_ssd/main.c]: reverted due to misunderstanding

File: inference-engine/ie_bridges/c/samples/hello_classification/README.md (hunk 2 above)
Comment (generalova-kate) [reply inference-engine/ie_bridges/c/samples/hello_classification/README.md:54]: Thanks for info. I added it to README as Validated image field to table

File: inference-engine/ie_bridges/c/samples/hello_nv12_input_classification/main.c (hunk 5 above)
Comment (generalova-kate) [reply inference-engine/ie_bridges/c/samples/hello_nv12_input_classification/main.c]: updated to bool

**Review by mryzhov — APPROVED** [review]

File: inference-engine/ie_bridges/c/samples/hello_classification/README.md [hunk 6]
```diff
@@ -1,31 +1,103 @@
...
+[ie_network_get_input_name]:https://docs.openvinotoolkit.org/latest/ie_c_api/group__Network.html#ga36b0c28dfab6db2bfcc2941fd57fbf6d
+[ie_network_set_input_precision]:https://docs.openvinotoolkit.org/latest/ie_c_api/group__Network.html#gadd99b7cc98b3c33daa2095b8a29f66d7
+[ie_network_get_output_name]:https://docs.openvinotoolkit.org/latest/ie_c_api/group__Network.html#ga1feabc49576db24d9821a150b2b50a6c
+[ie_network_get_output_precision]:https://docs.openvinotoolkit.org/latest/ie_c_api/group__Network.html#gaeaa7f1fb8f56956fc492cd9207235984
+[ie_core_load_network]:https://docs.openvinotoolkit.org/latest/ie_c_api/group__Core.html#ga318d4b0214b8a3fd33f9e44170befcc5
+[ie_exec_network_create_infer_request]:https://docs.openvinotoolkit.org/latest/ie_c_api/group__ExecutableNetwork.html#gae72247391c1429a18c367594a4b7db9f
+[ie_blob_make_memory_from_preallocated]:https://docs.openvinotoolkit.org/latest/ie_c_api/group__Blob.html#ga7a874d46375e10fa1a7e8e3d7e1c9c9c
+[ie_infer_request_set_blob]:https://docs.openvinotoolkit.org/latest/ie_c_api/group__InferRequest.html#ga891c2d475501bba761148a0c3faca196
```
Comment (ilya-lavrenov) [inline inference-engine/ie_bridges/c/samples/hello_classification/README.md:97]: I suppose such hardcoded links will be obsolete after new generation of docs, is not it?

CC @ntyukaev

Comment (generalova-kate) [reply inference-engine/ie_bridges/c/samples/hello_classification/README.md:97]: DO NOT MERGE until @ntyukaev provide a decision for C API links.

Comment (generalova-kate) [reply inference-engine/ie_bridges/c/samples/hello_classification/README.md:96]: We decided to merge changes as is and update links after changes in doc (PR @ntyukaev).

**Pull request merged**